### PR TITLE
OAI-PMH: export for DCAT-AP

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/inveniosoftware/invenio-rdm-records/issues/1234

### CODE REVIEW

File: invenio_rdm_records/resources/serializers/datacite/schema.py [hunk 1]
```diff
@@ -409,11 +409,11 @@ def get_subjects(self, obj):
...
         for subject in subjects:
-            sub_text = subject.get("subject")
-            if sub_text:
-                serialized_subjects.append({"subject": sub_text})
+            _id = subject.get("id")
+            if _id:
+                ids.append(_id)
             else:
-                ids.append(subject.get("id"))
+                serialized_subjects.append({"subject": subject.get("subject")})
```
Comment (rekt-hard) [inline invenio_rdm_records/resources/serializers/datacite/schema.py:416]: Will this access always evaluate to something to be used or should there be a check/default value?

**Review by rekt-hard — APPROVED** [review]
LGTM :)

File: invenio_rdm_records/resources/serializers/datacite/schema.py (hunk 1 above)
Comment (anikachurilova) [reply invenio_rdm_records/resources/serializers/datacite/schema.py:416]: The subjects list can be missing (which is handled on line 406). Bus as long as subjects is present, the subject inside is mandatory (it's either we add subjects or we don't)

**Review by jrcastro2 — APPROVED** [review]

File: invenio_rdm_records/oai.py [hunk 2]
```diff
@@ -40,6 +41,16 @@ def oai_marcxml_etree(pid, record):
     return etree.fromstring(MARCXMLSerializer().serialize_object(record["_source"]))
 
 
+def oai_dcat_etree(pid, record):
+    """OAI DCAT-AP format for OAI-PMH.
+
+    It assumes that record is a search result.
+    """
+    return etree.fromstring(
+        DCATSerializer().serialize_object(record["_source"]).encode(encoding="utf-8")
```
Comment (ppanero) [inline invenio_rdm_records/oai.py:50]: 🚀 very nice, I opened and issue to do this in other functions that don't 💪  https://github.com/inveniosoftware/invenio-rdm-records/issues/1255

**Review by ppanero — APPROVED** [review]

**Pull request merged**